<launch>
    <arg name="offset_distance"    default="4"/>
    <arg name="ground_thresh"      default="0.4"/>
    <arg name="down_size"          default="0.6"/><!-- 2 5-->
    <arg name="num_knn"            default="30"/>
    <!-- EC param-->
    <arg name="min_cluster"        default="3"/>
    <arg name="max_cluster"        default="100"/>
    <arg name="dist_ratio" default="3.0"/>
    <!-- MPL param-->
    <arg name="dt" default="0.2"/>
    <arg name="v_max" default="2.0"/>
    <arg name="a_max" default="1.0"/>
    <arg name="yaw_max" default="-1.0"/>
    <arg name="u_max" default="10.0"/>
    <arg name="u_yaw" default="3.0"/>
    <arg name="num" default="1"/>
    <arg name="max_num" default="-1"/>

    <arg name="path_min"           default="1.5"/><!-- 2.5 7.0-->
    <arg name="path_max"           default="7.0"/><!-- 2.5 7.0-->
    <arg name="octomap_resolution" default="0.3"/>
    <arg name="pub_octomap"        default="true"/>
    <arg name="pub_candidate"      default="true"/>
    <arg name="correction"      default="false"/>

    <param name="/offset_distance"    value="$(arg offset_distance)" />
    <param name="/ground_thresh"      value="$(arg ground_thresh)" />
    <param name="/down_size"          value="$(arg down_size)" />
    <param name="/num_knn"            value="$(arg num_knn)" />
    <param name="/min_cluster"        value="$(arg min_cluster)" />
    <param name="/max_cluster"        value="$(arg max_cluster)" />
    <param name="/path_min"           value="$(arg path_min)" />
    <param name="/path_max"           value="$(arg path_max)" />
    <param name="/octomap_resolution" value="$(arg octomap_resolution)" />
    <param name="/pub_octomap"        value="$(arg pub_octomap)" />
    <param name="/pub_candidate"      value="$(arg pub_candidate)" />
    <param name="/correction"      value="$(arg correction)" />

    <param name="dt" value="$(arg dt)"/>
    <param name="v_max" value="$(arg v_max)"/>
    <param name="a_max" value="$(arg a_max)"/>
    <param name="u_max" value="$(arg u_max)"/>
    <param name="yaw_max" value="$(arg yaw_max)"/>
    <param name="u_yaw" value="$(arg u_yaw)"/>
    <!--<param name="t_max" value="$(eval dt*10)"/>-->
    <param name="t_max" value="-1"/>
    <param name="use_prior" value="false"/>
    <param name="use_acc" value="false"/>
    <param name="use_jrk" value="true"/>
    <param name="num" value="2"/>
    <param name="w" value="10000"/>
    <param name="epsilon" value="2"/>
    <param name="max_num" value="$(arg max_num)"/>
    <param name="robot_r" value="0.5"/>
    <!--xterm -e *run* args-->
    <node pkg="n_cpp" type="n_path" name="n_path" output="screen" launch-prefix="gdb -ex run --args"/>
</launch>
